src/commands/setup.rs

```diff
@@ -77,29 +81,36 @@ impl Setup {
                     //Configure Bridge and veth_pairs
                     let status_block = network::core::Core::bridge_per_podman_network(
                         per_network_opts,
-                        network,
+                        &network,
                         &self.network_namespace_path,
                     )?;
                     response.insert(net_name.to_owned(), status_block);
 
-                    let id_network_hash = CoreUtils::create_network_hash(net_name, MAX_HASH_SIZE);
+                    let id_network_hash = CoreUtils::create_network_hash(&net_name, MAX_HASH_SIZE);
                     let sn = SetupNetwork {
-                        net: (*network).clone(),
+                        net: network.clone(),
                         network_hash_name: id_network_hash.clone(),
                     };
                     firewall_driver.setup_network(sn)?;
-
                     let port_bindings = network_options.port_mappings.clone();
                     match port_bindings {
                         None => {}
                         Some(i) => {
+                            let container_ips =
+                                per_network_opts.static_ips.as_ref().ok_or_else(|| {
+                                    std::io::Error::new(
+                                        std::io::ErrorKind::Other,
+                                        "no container ip provided",
+                                    )
+                                })?;
+                            // let container_ip = container_ips[idx];
                             let spf = SetupPortForward {
-                                net: (*network).clone(),
+                                net: network.clone(),
                                 container_id: network_options.container_id.clone(),
                                 port_mappings: i,
-                                network_name: (*net_name).clone(),
+                                network_name: net_name,
                                 network_hash_name: id_network_hash,
-                                options: (*per_network_opts).clone(),
+                                container_ip: container_ips[idx],
```

Comment:
The loop order does not make sense! You use the index from the networks loop but the static_ips are per network, len(static_ips) == len(subnets). What you have to do for your port forwarding logic is loop over the static ips and take the first ipv4 (if there is one) and the first ipv6 (if there is one). If there are only ipv4 you only forward to the first ipv4 not all ips. The same for ipv6.

The same for the teardown logic